packages/website/src/components/modals/Modal.tsx

```diff
@@ -12,24 +12,46 @@ interface ModalProps {
 }
 
 function Modal(props: ModalProps): JSX.Element {
+  const closeOnEscapeKeyDown = useCallback(
+    (e: KeyboardEvent): void => {
+      if (e.key === 'Escape' || e.keyCode === 27) {
+        props.onClose();
+      }
+    },
+    [props.onClose],
+  );
+
+  useEffect(() => {
```

Comment:
no need to declare this outside the effect if it's only used inside the effect

```suggestion
  useEffect(() => {
    const closeOnEscapeKeyDown = (
      (e: KeyboardEvent): void => {
        if (e.key === 'Escape' || e.keyCode === 27) {
          props.onClose();
        }
      },
    );
```